core/transfer/transfer.go

```diff
@@ -127,6 +128,48 @@ func WithProgress(f ProgressFunc) Opt {
 	}
 }
 
+type FetcherConfig struct {
```

Comment:
isn't `remotes/resolver.go` a better place to define these?

Reply:
This was not a possibility as `core/remotes` imports `core/transfer` and I needed to pass these options to `transfer.ImageFetcher`. Creating an import cycle.

Reply:
https://github.com/azr/containerd/blob/dcd4b1234d11c43066fc7e81718744819b5a52f8/core/transfer/transfer.go#L38-L42

Reply:
After thinking a bit, I'm feeling like this is okay like this, but lmk !

Reply:
Feel free to ignore my suggestion here.

My thinking was we define the new FetcherOpt directly in remotes/resolver.go instead of in transfer.
So the Resolver interface changes to
```
type Resolver interface {

-       Fetcher(ctx context.Context, ref string, opts ...transfer.FetcherOpt) (Fetcher, error)
+       Fetcher(ctx context.Context, ref string, opts ...FetcherOpt) (Fetcher, error)
```
So, remotes does not import transfer (importing fetcherOpts from transfer felt a bit weird)
And in the transfer package, the ImageFetcher interface can be changed to:
```
 type ImageFetcher interface {
        ImageResolver
 
-       Fetcher(ctx context.Context, ref string, opts ...FetcherOpt) (Fetcher, error)
+       Fetcher(ctx context.Context, ref string, opts ...remotes.FetcherOpt) (Fetcher, error)
```
There should not be any cyclic dependencies since package resolver or package docker will not have a dependency on transfer, but transfer will depend on package transfer.

Reply:
Ah, very good point, this is way better, thanks !